spec/services/hearings_creator_spec.rb

```diff
@@ -215,6 +250,105 @@
+                                                                               jurisdiction_type: "MAGISTRATES",
+                                                                               case_urn: "12345",
+                                                                               defendant: defendant_one,
+                                                                               court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
+        expect(Sqs::PublishHearing).to receive(:call).once.with(hash_including(shared_time: "2018-10-25 11:30:00",
+                                                                               jurisdiction_type: "MAGISTRATES",
+                                                                               case_urn: "12345",
+                                                                               defendant: defendant_two,
+                                                                               court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
+        create_hearings
+      end
+    end
+
+    context "with two defendants one with linked LAA reference, one without" do
+      let(:prosecution_case_array) { nil }
+      let(:defendant_case_array) { [defendant_case_one, defendant_case_two] }
+
+      it "calls the Sqs::PublishHearing service twice" do
+        allow(LaaReference).to receive(:find_by).with(defendant_id: "dd22b110-7fbc-3036-a076-e4bb40d0a666", linked: true).and_return(defendant_one)
+        allow(LaaReference).to receive(:find_by).with(defendant_id: "ad22b110-7fbc-3036-a076-e4bb40d0a667", linked: true).and_return(nil)
+
+        expect(Sqs::PublishHearing).to receive(:call).once.with(hash_including(shared_time: "2018-10-25 11:30:00",
+                                                                               jurisdiction_type: "MAGISTRATES",
+                                                                               case_urn: "12345",
+                                                                               defendant: defendant_one,
+                                                                               court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
+
+        create_hearings
+      end
+    end
+
+    context "with no defendant cases" do
+      let(:prosecution_case_array) { nil }
+      let(:defendant_case_array) { nil }
+
+      it "does not call the Sqs::PublishHearing service" do
+        expect(Sqs::PublishHearing).not_to receive(:call)
+        create_hearings
+      end
+    end
+
+    context "with a defendant that has no LAA reference" do
+      let(:prosecution_case_array) { nil }
+
+      it "does not call the Sqs::PublishHearing service" do
+        allow(LaaReference).to receive(:find_by).and_return(nil)
+
+        expect(Sqs::PublishHearing).not_to receive(:call)
+        create_hearings
+      end
+    end
+  end
+
+  context "when hearing had both a prosecution case and an applicationt to court" do
```

Comment:
Might not be necessary to test this combination, as the two procedures are totally independent from one another.

Reply:
I thought it makes sense to test, because it is still the HearingCreator's responsibility to split the hearing into prosecution cases and court applications